internal/pkg/cli/task_run.go

```diff
@@ -371,6 +379,42 @@ func (o *runTaskOpts) Validate() error {
 		}
 	}
 
+	if len(o.secrets) > 0 {
+		if err := o.validateSecrets(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateSecrets() error {
```

Comment:
nit: I feel like this is less about "validating" but more about "asking for confirmation", reflecting on function name perhaps we can have:
```suggestion
func (o *runTaskOpts) confirmSecretsAccess() error {
```